src/realm/object-store/sync/sync_manager.cpp

```diff
@@ -446,6 +446,30 @@ void SyncManager::remove_user(const std::string& user_id)
     }
 }
 
+SyncManager::~SyncManager()
+{
+    std::vector<std::shared_ptr<SyncSession>> current_sessions;
+
+    {
+        // We need to release the session lock and copy the sessions to another
+        // collection because the mutex will be acquited and m_sessions will be
+        // manipulated in unregister_session
+        std::lock_guard<std::mutex> lock(m_session_mutex);
```

Comment:
There should be no need to do any locking in the destructor, as any sort of concurrent access would be UB anyway.